From 03ca239cbab4262d84bdb6096abd2b41322f9c3d Mon Sep 17 00:00:00 2001
From: Daniel Mueller <deso@posteo.net>
Date: Mon, 22 Oct 2018 19:14:42 -0700
Subject: [PATCH 4/4] Use Wide formatting style for separating type parameter
 constraints

While the "Compressed" punctuation style is suitable for most occasions, the
formatting of generic parameter constraints looks off without spaces.
Unfortunately, however, it cannot be configured on its own.
So with this change we force the "Wide" style to be used for those
cases.
---
 src/tools/rustfmt/src/types.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/tools/rustfmt/src/types.rs b/src/tools/rustfmt/src/types.rs
index dd7f60..de59f0 100644
--- a/src/tools/rustfmt/src/types.rs
+++ b/src/tools/rustfmt/src/types.rs
@@ -768,7 +768,7 @@ fn join_bounds(
 ) -> Option<String> {
     // Try to join types in a single line
     let joiner = match context.config.type_punctuation_density() {
-        TypeDensity::Compressed => "+",
+        TypeDensity::Compressed |
         TypeDensity::Wide => " + ",
     };
     let type_strs = items
-- 
2.18.1

